src/hex_http.erl

```diff
@@ -18,10 +18,9 @@
 -spec request(hex_core:config(), method(), URI :: binary(), headers(), body()) ->
     {ok, {status(), headers(), binary()}} | {error, term()}.
 request(Config, Method, URI, Headers, Body) when is_binary(URI) and is_map(Headers) ->
-    Adapter = maps:get(http_adapter, Config),
+    {Adapter, AdapterConfig} = maps:get(http_adapter, Config, {hex_http_httpc, #{}}),
```

Comment:
I believe this should work (I haven't tested it), could you give it a try? Other than, the PR looks good to.

@ericmj I'd be ok with making the breaking change and shipping v0.8, but if the deprecation is as simple as this, I'd go with it:

```suggestion
    case maps:get(http_adapter, Config, {hex_http_httpc, #{}}) of
      {Adapter, AdapterConfig} ->
        {Adapter, AdapterConfig};

      %% TODO: remove in v0.9
      Adapter when is_atom(Adapter) ->
        AdapterConfig = maps:get(http_adapter_config, Config, #{}),
        io:format("[hex_http] setting #{http_adapter => Module, http_adapter_config => Map} is deprecated in favour of #{http_adapter => {Module, Map}}~n"),
        {Adapter, AdapterConfig}
    end
```

Reply:
@wojtekmach I tested the example calls from the README and it works. I will commit the suggested changes regarding the deprecation in a bit :)